beacon_node/beacon_chain/src/light_client_optimistic_update_verification.rs

```diff
@@ -110,30 +85,21 @@ impl<T: BeaconChainTypes> VerifiedLightClientOptimisticUpdate<T> {
             None => return Err(Error::SigSlotStartIsNone),
         }
 
-        // check if we can process the optimistic update immediately
-        // otherwise queue
-        let canonical_root = light_client_optimistic_update
-            .attested_header
-            .beacon
-            .canonical_root();
-
-        if canonical_root != head_block.message().parent_root() {
-            return Err(Error::UnknownBlockParentRoot(canonical_root));
-        }
-
-        let optimistic_update =
-            LightClientOptimisticUpdate::new(&chain.spec, head_block, &attested_state)?;
+        let latest_optimistic_update = chain
+            .lightclient_server_cache
+            .get_latest_optimistic_update()
+            .ok_or(Error::FailedConstructingUpdate)?;
 
         // verify that the gossiped optimistic update is the same as the locally constructed one.
-        if optimistic_update != light_client_optimistic_update {
+        if latest_optimistic_update != rcv_optimistic_update {
             return Err(Error::InvalidLightClientOptimisticUpdate);
         }
 
-        *latest_seen_optimistic_update = Some(light_client_optimistic_update.clone());
-
+        let parent_root = rcv_optimistic_update.attested_header.beacon.parent_root;
         Ok(Self {
-            light_client_optimistic_update,
-            parent_root: canonical_root,
+            light_client_optimistic_update: rcv_optimistic_update,
+            // TODO: why is the parent_root necessary here?
+            parent_root,
```

Comment:
Looks like the `Ok` output of this function is only used for logging below, and we don't use any of these output vallues, so we could potentially refactor this to just return `Result<(), Error>`

https://github.com/sigp/lighthouse/blob/ba891e1fed8379f1410f50ce56eccc05f25daac2/beacon_node/network/src/network_beacon_processor/gossip_methods.rs#L1751-L1759

Reply:
Sure, let's do that in a future pr dealing with gossip